pennylane/ops/functions/dot.py

```diff
@@ -77,7 +77,7 @@ def dot(coeffs: Sequence[float], ops: Sequence[Operator], pauli=False):
     if pauli:
         return _pauli_dot(coeffs, ops)
 
-    operands = [qml.s_prod(coeff, op) for coeff, op in zip(coeffs, ops)]
+    operands = [op if coeff == 1 else qml.s_prod(coeff, op) for coeff, op in zip(coeffs, ops)]
```

Comment:
For readability. Having `SProd` doesn't reduce performance.

It also simplifies the comparison of operators, because:
```pycon
>>> qml.equal(qml.s_prod(1, qml.PauliX(0)), qml.PauliX(0))
False
```